### PR TITLE
Merge develop into master

### PR DESCRIPTION
**Pull request by marvas**
Contains cleanup, bugfix and more tests.

### CODE REVIEW

File: Heine.Mvc.ActionFilters.Tests/Services/ObfuscatorServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            var obfuscationGraph = obfuscationService.TypeObfuscationGraphs;
+            var documentGraph = obfuscationGraph[typeof(Document)];
+            var personGraph = obfuscationGraph[typeof(Person)];
+
+            // Assert
+            obfuscationGraph.Should().NotBeEmpty();
+            documentGraph.Should().NotBeEmpty();
+            personGraph.Should().NotBeEmpty();
+            documentGraph.Count.Should().Be(27);
+            documentGraph.FirstOrDefault(s => s.Equals("Content")).Should().NotBeNullOrEmpty();
```
Comment (sandves) [inline Heine.Mvc.ActionFilters.Tests/Services/ObfuscatorServiceTests.cs]: Consider to change the assertions to the following format for better readability:
```suggestion
            documentGraph.Should.Contain(s => s.Equals("Content"));
```

File: Heine.Mvc.ActionFilters.Tests/Services/ObfuscatorServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+            documentGraph.FirstOrDefault(s => s.Equals("Children[*].Content")).Should().NotBeNullOrEmpty();
+            documentGraph.FirstOrDefault(s => s.Equals("Children[*].Owner.Mobil")).Should().NotBeNullOrEmpty();
+            documentGraph.FirstOrDefault(s => s.Equals("Children[*].Owner.BankAccNo")).Should().NotBeNullOrEmpty();
+            documentGraph.FirstOrDefault(s => s.Equals("Children[*].Owner.CreditDetail")).Should().NotBeNullOrEmpty();
+            documentGraph.FirstOrDefault(s => s.Equals("Children[*].Owner.Remarks[*]")).Should().NotBeNullOrEmpty();
+            documentGraph.FirstOrDefault(s => s.Equals("Children[*].Persons[*].Mobil")).Should().NotBeNullOrEmpty();
+            documentGraph.FirstOrDefault(s => s.Equals("Children[*].Persons[*].BankAccNo")).Should().NotBeNullOrEmpty();
+            documentGraph.FirstOrDefault(s => s.Equals("Children[*].Persons[*].Remarks[*]")).Should().NotBeNullOrEmpty();
+            documentGraph.FirstOrDefault(s => s.Equals("Children[*].Persons[*].CreditDetail")).Should().NotBeNullOrEmpty();
+            personGraph.Count.Should().Be(6);
```
Comment (sandves) [inline Heine.Mvc.ActionFilters.Tests/Services/ObfuscatorServiceTests.cs]:
```suggestion
            personGraph.Should().HaveCount(6);
```

**Review by sandves — CHANGES_REQUESTED** [review]
Added a couple of suggestions to simplify the assertions.

File: Heine.Mvc.ActionFilters.Tests/Services/ObfuscatorServiceTests.cs (hunk 1 above)
Comment (marvas) [reply Heine.Mvc.ActionFilters.Tests/Services/ObfuscatorServiceTests.cs]: Good tip. Change has been implemented.

File: Heine.Mvc.ActionFilters.Tests/Services/ObfuscatorServiceTests.cs (hunk 2 above)
Comment (marvas) [reply Heine.Mvc.ActionFilters.Tests/Services/ObfuscatorServiceTests.cs]: This has been implemented.

**Review by sandves — APPROVED** [review]

**Pull request merged**